src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -93,8 +93,15 @@ class IpexCommunicationService extends AgentService {
       );
     }
 
+    if (grantNoteRecord.linkedRequest) {
```

Comment:
in this function we are storing a pending connection twice:

```js
if (grantNoteRecord.linkedRequest) {
  await this.operationPendingStorage.save({
    id: grantNoteRecord.linkedRequest.current,
    recordType: OperationPendingRecordType.ExchangeReceiveCredential,
  });
}
```
 and:

```js
const pendingOperation = await this.operationPendingStorage.save({
  id: op.name,
  recordType: OperationPendingRecordType.ExchangeReceiveCredential,
});
```
is there a case where `grantNoteRecord.linkedRequest.current` and `op.name` are the same? we should avoid creating the pendingOperation twice if so